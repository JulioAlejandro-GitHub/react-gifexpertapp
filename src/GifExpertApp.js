
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const  GifExpertApp = ( {defaultCatergories=[]} ) => {
  //const categories = ['one', 'two', 'tree'];
  const [categories, setCategories] = useState(defaultCatergories);

  /*
  const handleAdd = () => {
    //setCategories( [...categories, 'four'] );
    setCategories( cate => [...cate, 'four'] );
  }
*/
  return (
    <>
      <h2>Gifs Expert App</h2>
      <hr/>
      <AddCategory setCategories={setCategories} />
      <ol>
        {
          categories.map( (category, i) => (
            <GifGrid 
              category={category}
              key={i}
            />
          ))
        }
      </ol>
    </>
  );
}

GifExpertApp.propTypes = {
  defaultCatergories: PropTypes.array.isRequired
}
GifExpertApp.defaultProps = {
  defaultCatergories : ["Cats BD"]
}

export default GifExpertApp;
