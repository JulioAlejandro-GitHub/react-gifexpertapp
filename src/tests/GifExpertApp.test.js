import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";


describe('Test <GifExpertApp />', () => {



    test('Test: Componente [GifExpertApp]', () => {
        const wrapper = shallow(
            <GifExpertApp defaultCatergories={['dogs']}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Test: Componente [GifExpertApp] listar categorias', () => {
        const categories = ['Goku', 'mickey'];
        const wrapper = shallow(<GifExpertApp defaultCatergories={categories}/>);

        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
});