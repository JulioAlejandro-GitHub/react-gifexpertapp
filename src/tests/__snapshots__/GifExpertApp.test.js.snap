// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp /> Test: Componente [GifExpertApp] 1`] = `
<Fragment>
  <h2>
    Gifs Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="dogs"
      key="0"
    />
  </ol>
</Fragment>
`;
